pkg/collector/corechecks/snmp/session.go

```diff
@@ -0,0 +1,51 @@
+package snmp
+
+import (
+	"github.com/soniah/gosnmp"
+	"time"
+)
+
+type sessionAPI interface {
+	Configure(config snmpConfig)
+	Connect() error
+	Close() error
+	Get(oids []string) (result *gosnmp.SnmpPacket, err error)
+	GetBulk(oids []string) (result *gosnmp.SnmpPacket, err error)
+}
+
+type snmpSession struct {
+	gosnmpInst gosnmp.GoSNMP
+}
+
+func (s *snmpSession) Configure(config snmpConfig) {
+	s.gosnmpInst = gosnmp.GoSNMP{
+		Target:    config.IPAddress,
+		Port:      config.Port,
+		Community: config.CommunityString,
+		Version:   config.SnmpVersion,
+		// TODO: implement following configs
+		Timeout:            time.Duration(2) * time.Second,
+		Retries:            3,
+		ExponentialTimeout: true,
+		MaxOids:            100,
+	}
+}
+
+func (s *snmpSession) Connect() error {
+	return s.gosnmpInst.Connect()
+}
+
+func (s *snmpSession) Close() error {
+	return s.gosnmpInst.Conn.Close()
+}
+
+func (s *snmpSession) Get(oids []string) (result *gosnmp.SnmpPacket, err error) {
+	return s.gosnmpInst.Get(oids)
```

Comment:
what if you call `Get` before `Connect` ?

Reply:
That will trigger an error like:

```
error: Get() err: &GoSNMP.Conn is missing. Provide a connection or use Connect()
```

Reply:
Made some improvement to error handling: https://github.com/DataDog/datadog-agent/pull/6749/commits/7a7cec09a6cabd0af4d9c5e998411428af1431b6